devito/ir/clusters/cluster.py

```diff
@@ -35,8 +35,11 @@ class Cluster(object):
     """
 
     def __init__(self, exprs, ispace, dspace, guards=None, properties=None):
-        self._exprs = tuple(ClusterizedEq(i, ispace=ispace, dspace=dspace)
-                            for i in as_tuple(exprs))
+        self._exprs = tuple(
+            ClusterizedBreak(i, ispace=ispace, dspace=dspace) if i.is_Break
```

Comment:
ClusterizedBreak looks like a hack

I wonder whether we're actually getting the abstractions right (@rhodrin )

Reply:
btw, if it's an Eq subtype, why is it even necessary to have a separate type of Cluster?

Reply:
Yeah, not sure, will take a look.